float64 left_pos
float64 right_pos
float64 left_neg
float64 right_neg
int64 id
